waku/persistence/store.go

```diff
@@ -224,7 +224,7 @@ func (d *DBStore) cleanOlderRecords(ctx context.Context) error {
 	// Limit number of records to a max N
 	if d.maxMessages > 0 {
 		start := time.Now()
-		sqlStmt := `DELETE FROM message WHERE id IN (SELECT id FROM message ORDER BY receiverTimestamp DESC LIMIT -1 OFFSET $1)`
+		sqlStmt := `DELETE FROM message WHERE id IN (SELECT id FROM message ORDER BY receiverTimestamp DESC OFFSET $1)`
```

Comment:
This does not work in SQLite. The LIMIT keyword is required when specifying an OFFSET.   If postgresql does not allow negative limits, then we need to extract this query, similar to how it was done with CreateTable in here https://github.com/waku-org/go-waku/blob/16ec22596ef3e4b1a65af1ea1f2a0429847c9f7a/waku/persistence/postgres/postgres.go#L59-L66 which has db specific implementations

Reply:
This sounds like a bug!